src/java/org/apache/cassandra/config/CassandraRelevantProperties.java

```diff
@@ -424,8 +424,9 @@ public enum CassandraRelevantProperties
     SAI_INTERSECTION_CLAUSE_LIMIT("cassandra.sai.intersection_clause_limit", "2"),
     /** Latest version to be used for SAI index writing */
     SAI_LATEST_VERSION("cassandra.sai.latest_version", "aa"),
-    SAI_MAX_FROZEN_TERM_SIZE("cassandra.sai.max_frozen_term_size_kb", "5"),
-    SAI_MAX_STRING_TERM_SIZE("cassandra.sai.max_string_term_size_kb", "1"),
+    SAI_MAX_FROZEN_TERM_SIZE("cassandra.sai.max_frozen_term_size", "5KiB"),
+    SAI_MAX_STRING_TERM_SIZE("cassandra.sai.max_string_term_size", "1KiB"),
+    SAI_MAX_VECTOR_TERM_SIZE("cassandra.sai.max_vector_term_size", "16KiB"),
```

Comment:
We're allowing a max of 8k dimension which would be 32KB in Astra, fwiw

Reply:
Besides rising the default, perhaps we might want to express the term size as the number of dimensions, given that we only support vectors of floats? Like in `cassandra.sai.max_vector_dimensions`

Reply:
imo KB makes more sense so it aligns w/ the other term size limits

Reply:
I have set it to 32KiB for now, but there is the potential (later) to remove the term size restriction entirely. This is only in place because the in-memory index builders had a size limitation. The numeric index now uses a trie for the in-memory, and we have a ticket to move the literal index to use the same. This would mean we could remove these and replace them with guardrails instead.